protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py

```diff
@@ -17,10 +17,25 @@ def __init__(self, contracts_controller: ContractsController):
 
     @property
     def name(self) -> str:
-        return "deploy"
+        return "deploy_tp"
 
-    def _build(self) -> Callable[[Any], Any]:
-        return self.deploy_prepared
+    def _build(self) -> Callable[[list[int], int, int], Any]:
+        return self.deploy_prepared_tp
+
+    def deploy_prepared_tp(
+        self,
+        constructor_calldata: list[int],
+        contract_address: int,
+        class_hash: int,
+    ):
+        return self.deploy_prepared(
+            PreparedContract(
+                constructor_calldata=constructor_calldata,
+                contract_address=contract_address,
+                class_hash=class_hash,
+                salt=0,
+            )
+        )
```

Comment:
Shouldn't `deploy_prepared` raise `TransactionRevertException` instead of `CheatcodeException` so it is converted into an error with proper code?

I'm not 100% sure about this but I think it makes sense to make `deploy` cheatcodes error handleable by user.

Reply:
good idea [a568d73](https://github.com/software-mansion/protostar/pull/1716/commits/a568d73a9ef298d4c8eb6dac3342ea744937792d)